week2/index.js

```diff
@@ -0,0 +1,96 @@
+const fs = require('fs');
+const TODOS_FILENAME = process.env.TODOS_FILENAME || 'todo.json';
+let myTodoList = [];
+
+const readTodosFile = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            console.log(`Successfully read the file ${filename}`);
+            resolve(data);
+        });
+    });
+};
+const writeTodosFile = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(filename, dataAsString, (err) => {
+            if (err) {
+                if (err.code === "ENOENT") {
```

Comment:
Don't copy-paste without thinking - you won't get a file not found error when writing, because writing creates the file. 😄

Reply:
it,s not about copy-past.  just I do'nt know what is the  err I have to handel. so but the same eer handling. 
you mean like that:  
if (err) {
                reject(err);
            }
            resolve(`The < ${filename}> file has been saved!`);